test/functional/mining_signet.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test signet mining RPCs
+
+- getmininginfo
+- getblocktemplate proposal mode
+- submitblock"""
+
+from decimal import Decimal
+
+from test_framework.blocktools import (
+    create_coinbase,
+)
+from test_framework.messages import (
+    CBlock,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+
+private_key  = "ZFNp4W4HSatHDc4zsheMsDWgUg2uiB5PtnY2Y2hUCNCkQiF81KyV"
+pubkey       = "02c72b24ae2ee333f2da24aea66ce4338c01db8f26d0cc96f586d77edcb5238a4f"
+address      = "sb1qghzjzc0jvkjvvvymxnadmjjpu2tywmvagduwfj"
+blockscript  = "5121" + pubkey + "51ae" # 1-of-1 multisig
+genesisnonce = 405557
+
+def assert_template(node, block, expect, rehash=True):
+    if rehash:
+        block.hashMerkleRoot = block.calc_merkle_root()
+    rsp = node.getblocktemplate(template_request={'data': block.serialize().hex(), 'mode': 'proposal', 'rules': ['segwit']})
+    assert_equal(rsp, expect)
+
+
+class SigMiningTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.chain = "signet"
+        self.num_nodes = 2
+        self.setup_clean_chain = True
+        shared_args = ["-signet_blockscript=" + blockscript, "-signet_genesisnonce=" + str(genesisnonce), "-signet_seednode=localhost:1234"]
+        self.extra_args = [shared_args, shared_args]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def generate(self, node, count):
+        for i in range(count):
+            addr = node.getnewaddress()
+            block = node.getnewblockhex(addr)
+            signed = node.signblock(block)
+            node.grindblock(signed)
+
+    def run_test(self):
+        node = self.nodes[0]
+
+        # give the privkey to node 1 so it can sign
+        node.importprivkey(private_key)
+        self.log.info('Imported network private key')
+        self.log.info('address: %s, privkey: %s' % (address, node.dumpprivkey(address)))
```

Comment:
Instead of string format arguments use logging function parameters when using `self.log.info` :)

Reply:
You mean ` % (...)` → `, ...`? OK.